packages/contracts-ops/contracts/scripts/UpgradeCallBatches.s.sol

```diff
@@ -0,0 +1,236 @@
+            string memory domain = domainNames[i];
+            loadBeacons(domain);
+            loadImplementations(domain);
+            loadController(domain);
+            generateGovernanceCalls(domain);
+        }
+        writeCallBatch(recovery);
+    }
+
+    function setUp() internal {
+        __Config_initialize(configFile);
+        string memory outputFile = "upgradeActions.json";
+        __CallBatch_initialize(
+            localDomainName,
+            getDomainNumber(localDomainName),
+            outputFile,
+            true
+        );
+        networksArray = getNetworks();
+    }
+
+    function loadBeacons(string memory domain) internal {
+        bridgeTokenBeacon = address(bridgeTokenUpgrade(domain).beacon);
+        tokenRegistryBeacon = address(tokenRegistryUpgrade(domain).beacon);
+        bridgeRouterBeacon = address(bridgeRouterUpgrade(domain).beacon);
+        governanceRouterBeacon = address(
+            governanceRouterUpgrade(domain).beacon
+        );
+        governanceRouterProxy = address(governanceRouterUpgrade(domain).proxy);
+        for (uint256 i; i < networksArray.length; i++) {
+            if (!compareStrings(networksArray[i], domain)) {
+                replicaBeacon = address(
+                    replicaOfUpgrade(domain, networksArray[i]).beacon
+                );
+                break;
+            }
+            homeImpl = address(homeUpgrade(domain).beacon);
+        }
+        homeBeacon = address(homeUpgrade(domain).beacon);
+    }
+
+    function loadImplementations(string memory domain) internal {
+        bridgeTokenImpl = bridgeTokenUpgrade(domain).implementation;
+        tokenRegistryImpl = tokenRegistryUpgrade(domain).implementation;
+        bridgeRouterImpl = bridgeRouterUpgrade(domain).implementation;
+        governanceRouterImpl = governanceRouterUpgrade(domain).implementation;
+        // All replicas have the same implmentation
+        for (uint256 i; i < networksArray.length; i++) {
+            if (!compareStrings(networksArray[i], domain)) {
+                replicaImpl = replicaOfUpgrade(domain, networksArray[i])
+                    .implementation;
+                break;
+            }
+            homeImpl = homeUpgrade(domain).implementation;
+        }
+    }
+
+    function loadController(string memory domain) internal {
+        beaconController = address(getUpgradeBeaconController(domain));
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                       GOVERNANCE CALL GENERATORS
+    //////////////////////////////////////////////////////////////*/
+
+    function generateGovernanceCalls(string memory domain) internal {
+        title("BeaconController upgrade encoded calls for", domain);
+        console2.log(
+            "Domain Number: ",
+            vm.toString(uint256(getDomainNumber(domain)))
+        );
+        console2.log("Function signature: upgrade(address, address)");
```

Comment:
for synchronous functions which execute in seconds (or less), i think it's not super helpful to do a bunch of console.logging. 

for async scripts like deployments or otherwise sending transactions, logging is useful, but foundry has very informative logging built-in, so i also don't find it that helpful in that case

i feel like adding inline comments to the code is better (and easier to read) and the person running the script just knows that it's doing what the code does

Reply:
It's helpful to visually see the result of the script with the calldata of every batch nicely printed. Far easier than reading the json file alone. 

I can remove them, but personally they seem helpful.

Reply:
well, in this case, most of what's being logged is
- each domain (not much easier to read than JSON keys)
- a hard-coded reminder of the function signature & argument structure (we know this already) 
- the raw bytes (not really easier to read than JSON)

Reply:
maybe if it was something like 

```
"Home Beacon:", homeBeacon
"Home New Implementation:", homeImpl
```

then that would indeed be easier to read than JSON.

but again, what is the main purpose of logging that type of thing? it feels like it would be useful mainly for debugging while you're developing, to run the script and make sure it's hitting the domains you expect and the arguments you expect. 

but i would rather that we ensure that the script actually does what we expect via 
- robust code review 
- fork testing
- staging testing 
- maybe spot checking JSON a couple times until we gain confidence in the script

than by utilizing a solution that makes the code itself harder to read (imo)

Reply:
Gotcha. Removing